homeassistant/components/tuya/humidifier.py

```diff
@@ -71,7 +71,9 @@ def async_discover_device(device_ids: list[str]) -> None:
         entities: list[TuyaHumidifierEntity] = []
         for device_id in device_ids:
             device = hass_data.manager.device_map[device_id]
-            if description := HUMIDIFIERS.get(device.category):
+            if (description := HUMIDIFIERS.get(device.category)) and any(
+                code in device.status for code in description.dpcode
```

Comment:
Actually, now thinking about it, this approach isn't solid.

The `status` doesn't have to provide this. Hence the `find_dpcode` exists.